L1Trigger/L1THGCal/src/backend/HGCalMulticlusteringHistoImpl.cc

```diff
@@ -326,40 +458,69 @@ std::vector<GlobalPoint> HGCalMulticlusteringHistoImpl::computeThresholdSeeds( c
 
 
 std::vector<l1t::HGCalMulticluster> HGCalMulticlusteringHistoImpl::clusterSeedMulticluster(const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtrs,
-											   const std::vector<GlobalPoint> & seeds){
-
+                                                                                           const std::vector<std::pair<GlobalPoint, double> > & seeds){
 
     std::map<int,l1t::HGCalMulticluster> mapSeedMulticluster;
     std::vector<l1t::HGCalMulticluster> multiclustersTmp;
 
     for(auto & clu : clustersPtrs){
 
-
         int z_side = triggerTools_.zside(clu->detId());
 
-        double minDist = dr_;
-        int targetSeed = -1;
+        double minDist = drA_ + drB_*(2.3 - std::abs(clu->eta()) ) ;
 
-        for( unsigned int iseed=0; iseed<seeds.size(); iseed++ ){
+        std::vector<int> targetSeeds;
+        std::vector<double> targetSeedsEnergy;
 
-            if( z_side*seeds[iseed].z()<0) continue;
+        for( unsigned int iseed=0; iseed<seeds.size(); iseed++ ){
 
-            double d = this->dR(*clu, seeds[iseed]);
+            if( z_side*seeds[iseed].first.z()<0) continue;
```

Comment:
Instead of accessing the seed multiple time with `seeds[iseed]` you could just get it once. It would make the code clearer.